### PR TITLE
Support microservice

### PR DESCRIPTION
**Pull request by andreldsa**
<p><b>Feature/Bug description:</b> A new application that handles the incoming needs and bugs of the user.</p>

<p><b>Solution:</b> Creation of a separated app (microservice) under the same gcloud directory, using the AngularJS with Firebase as direct database.</p>

<p><b>TODO/FIXME:</b> n/a</p>

### CODE REVIEW

File: support/auth/userService.js [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    'use strict';
+
+    var support = angular.module("support");
+
+    support.service("UserService", function UserService($http, $q) {
+        var service = this;
+
+        var USER_URI = "/api/user";
+
+        service.deleteInstitution = function deleteInstitution(institutionKey) {
```
Comment (mayzabeel) [inline support/auth/userService.js]: remover funções que não são usadas no support

File: support/auth/messageService.js [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    var support = angular.module("support");
+
+    support.service("MessageService", function MessageService($mdToast, $mdDialog) {
+        var service = this;
+
+        var msg = {
+            "auth/email-already-in-use": "Email informado já está cadastrado.",
+            "auth/wrong-password": "Senha incorreta ou usuário não possui senha.",
+            "auth/user-not-found": "Usuário não existe.",
+            "Error! The institution has been deleted": "A instituição está inativa.",
```
[the remark this answers is not in the document]
Comment (andreldsa) [reply support/auth/messageService.js]: :+1:

File: support/auth/user.js [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+    if (!this.current_institution) {
+        this.changeInstitution();
+    }
+}
+
+var SENT = "sent";
+
+var USER = "USER";
+
+User.prototype.changeInstitution = function changeInstitution(institution) {
```
[the remark this answers is not in the document]
Comment (andreldsa) [reply support/auth/user.js]: :+1:

File: support/auth/user.js [hunk 4]
```diff
@@ -0,0 +1,167 @@
...
+    if (this.institutions && this.institutions.length > 0) {
+        institution = institution || this.institutions[0];
+        this.current_institution = _.find(this.institutions, {'key': institution.key});
+        changeProfileColor(this, institution);
+        window.localStorage.userInfo = JSON.stringify(this);
+    }
+};
+
+User.prototype.follow = function follow(institution) {
```
[the remark this answers is not in the document]
Comment (andreldsa) [reply support/auth/user.js]: :+1:

File: support/auth/user.js [hunk 5]
```diff
@@ -0,0 +1,167 @@
...
+    this.follows.push(institution);
+};
+
+User.prototype.unfollow = function unfollow(institution) {
```
[the remark this answers is not in the document]
Comment (andreldsa) [reply support/auth/user.js]: :+1:

File: support/auth/userService.js (hunk 1 above)
Comment (andreldsa) [reply support/auth/userService.js]: :+1:

File: support/auth/loginController.js [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+                .required(true)
+                .ok("Redefinir Senha")
+                .cancel("Cancelar");
+
+            $mdDialog.show(confirm).then(function(email) {
+                AuthService.resetPassword(email);
+            });
+        };
+
+        loginCtrl.goToLandingPage = function goToLandingPage() {
```
Comment (mayzabeel) [inline support/auth/loginController.js]: remover loginCtrl.goToLandingPage e loginCtrl.requestInvite

File: support/auth/login.html [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+          <div style="padding: 15px; font-size: 14px;" md-colors="{background: 'grey-300'}" hide-xs>
+            <a href="" style="text-decoration: none;" ng-click="loginCtrl.resetPassword()">
+              <b md-colors="{color: 'teal-500'}">Esqueceu a senha?</b>
+            </a>
+          </div>
+        </md-card>
+    </div>
+  </div>
+  
+  <signup return-action="loginCtrl.requestInvite" callback="loginCtrl.redirect"
```
Comment (mayzabeel) [inline support/auth/login.html]: remover

File: support/auth/loginController.js [hunk 8]
```diff
@@ -0,0 +1,78 @@
...
+
+    var support = angular.module("support");
+
+    support.controller("LoginController", function LoginController(AuthService, MessageService, $state, $mdDialog, 
+            $stateParams, $location, $window) {
+        var loginCtrl = this;
+            
+        loginCtrl.user = {};
+
+        loginCtrl.isRequestInvite = false;
```
Comment (mayzabeel) [inline support/auth/loginController.js]: remover

File: support/auth/login.html [hunk 9]
```diff
@@ -0,0 +1,43 @@
+<md-content layout-fill layout="row" layout-align="center center" md-colors="{background: 'teal-500'}">
+  <div ng-if="!loginCtrl.isRequestInvite" flex-xs="90" flex-sm="65" flex-md="65" flex-lg="30" flex-xl="35">
```
Comment (mayzabeel) [inline support/auth/login.html]: remover ng-if

Comment (mayzabeel) [inline support/auth/login.html]: aqui

File: support/auth/login.html (hunk 7 above)
Comment (mayzabeel) [reply support/auth/login.html]: aqui

File: support/auth/loginController.js (hunk 8 above)
Comment (mayzabeel) [inline support/auth/loginController.js]: aqui

File: support/auth/loginController.js (hunk 6 above)
Comment (mayzabeel) [inline support/auth/loginController.js]: remover loginCtrl.goToLandingPage e loginCtrl.requestInvite

File: support/auth/user.js [hunk 10]
```diff
@@ -0,0 +1,23 @@
+"use strict";
+
+function User(data) {
+    data = data || {};
+    _.extend(this, data);
+}
+
+var SENT = "sent";
```
Comment (Luiz-FS) [inline support/auth/user.js]: "SENT"

File: support/auth/userService.js [hunk 11]
```diff
@@ -0,0 +1,32 @@
...
+    'use strict';
+
+    var support = angular.module("support");
+
+    support.service("UserService", function UserService($http, $q) {
+        var service = this;
+
+        var USER_URI = "/api/user";
+
+        service.save = function save(patch) {
```
Comment (raonismaneoto) [inline support/auth/userService.js]: Do you use this function?

File: support/auth/authService.js [hunk 12]
```diff
@@ -0,0 +1,220 @@
...
+                return true;
+            }
+            return false;
+        };
+
+        service.save = function() {
+            $window.localStorage.userInfo = JSON.stringify(userInfo);
+        };
+
+        service.reload = function reload() {
```
Comment (Luiz-FS) [inline support/auth/authService.js]: Método não utilizado

File: support/index.html [hunk 13]
```diff
@@ -0,0 +1,78 @@
+<html lang="pt-br" >
+<head>
+  <meta name="viewport" charset="UTF-8" content="width=device-width, initial-scale=1">
+
+  <base href="/" />
+
+  <title>e-CIS | Suporte</title>
```
Comment (raonismaneoto) [inline support/index.html]: e-CIS or CIS?

File: support/support.js [hunk 14]
```diff
@@ -0,0 +1,183 @@
+(function() {
+    'use strict';
+
+    var support = angular.module('support', [
+        'ngMaterial',
+        'ui.router',
+        'ngAnimate',
+        'firebase',
```
Comment (raonismaneoto) [inline support/support.js]: @ruanres I think that variable comes from here..

File: support/firebaseService.js [hunk 15]
```diff
@@ -0,0 +1,29 @@
+(function () {
+    var support = angular.module('support');
+
+    support.service('FirebaseService', function ($firebaseArray, AuthService) {
+        var service = this;
+
+        var firebaseRef = firebase.database().ref();
```
[the remark this answers is not in the document]
Comment (andreldsa) [reply support/firebaseService.js]: Comes from the Firebase JS library.

File: support/auth/userService.js (hunk 11 above)
Comment (andreldsa) [reply support/auth/userService.js]: Removed

File: support/index.html (hunk 13 above)
Comment (andreldsa) [reply support/index.html]: done

File: support/support.js (hunk 14 above)
Comment (andreldsa) [reply support/support.js]: :+1:

File: support/firebaseService.js (hunk 15 above)
[the remark this answers is not in the document]
Comment (andreldsa) [reply support/firebaseService.js]: :+1:

**Pull request merged**